integration/integration_test.go

```diff
@@ -5040,11 +5040,12 @@ func testBPFSessionDifferentiation(t *testing.T, suite *integrationTestSuite) {
 
 	// Wait 10 seconds for both events to arrive, otherwise timeout.
 	timeout := time.After(10 * time.Second)
+outer:
 	for i := 0; i < 2; i++ {
 		select {
 		case <-doneCh:
 			if i == 1 {
-				break
+				break outer
 			}
```

Comment:
This if-block doesn't seem to serve a purpose and could be removed. The loop is going to exit the second time through.

Reply:
It took me a moment to understand it wants two events as well.

What about something like this (no need for a label either):

```go
	gotEvents := 0
	for {
		select {
		case <-doneCh:
			gotEvents++
		case <-timeout:
			dumpGoroutineProfile()
			require.FailNow(t, "Timed out waiting for client to finish interactive session.")
		}
		if gotEvents == 2 {
			break
		}
	}
```

Reply:
@codingllama Yeah, it isn't clear what the `2` means and requires looking at the preceding code. I like to use slices in this scenario. Also like to use `chan struct{}` instead of `chan bool` when the channel is meant to only be a signal so you don't have to reason about the difference between `true` and `false`.

```go
terms := []*Terminal{NewTerminal(250), NewTerminal(250)}
doneCh := make(chan struct{}, len(terms))
...
for _, term := range terms {
  go writeTerm(term) // assuming this should be a goroutines?
}

for _, term := range terms {
  select {
  case <-doneCh:
  case <-timeout:
     dumpGoroutineProfile()
     require.FailNow(t, "Timed out waiting for client to finish interactive session.")
  }
}
...
// away, test was successful
if len(sessionIDs) == len(terms) {
  return
}
```

Reply:
+1 for `chan struct{}`.

As others pointed out, I'm also confused by the fact that writeTerm is sync; could we ditch the channels then>

Reply:
If we do run `writeTerm` in a goroutine, maybe we should use a `ContextWithTimeout` for `client.SSH` (called by writeTerm) to reduce chance leaking go routines.

A side effect using select in a for loop is the second term kinda waits timeout twice. Not a big deal but may not be exactly what we wanted. I think we can either predefine a timer like `timer := time.NewTimer(timeout)` and use the same timer's chan in the loop. Or maybe use testify's `assert.Eventually` to check `len(doneCh) == len(terms)`.